swig/include/python/docs/gdal_operations_docs.i

```diff
@@ -24,6 +24,12 @@ yB : int
     The Y location (raster row) of the second point to check on the raster.
 zB : float
     The Z location (height) of the second point to check.
+xTerrainIntersection : int
```

Comment:
Should this use a `typemap`? I'm new to SWIG.
Reference: https://www.swig.org/Doc1.3/Python.html#Python_nn18

See `31.9.3 Using typemaps to return arguments`

Reply:
This is going a bit tricky to map to SWIG. Yes, we'd likely need a typemap per language. For Python, I guess we'd want to return a triple (is_visible, xInterection, yIntersection). 

So in the #ifdef SWIGPYTHON case, we'd probably want to have a wrapper like

void IsLineOfSightVisible(GDALRasterBandShadow *band,
                          int xA, int yA, double zA,
                          int xB, int yB, double zB,
                          bool* pbIntersects, int *xTerrainIntersection, int *yTerrainIntersection,
                          char** options = NULL)

and a (bool* pbIntersects, int *xTerrainIntersection, int *yTerrainIntersection) typemap. One of the closest example to take inspiration from would be the (int *pnxvalid, int *pnyvalid, int* pisvalid) typemap in swig/include/python/typemaps_python.i and used in swig/include/Band.i

For other languages, I would omit for now the int *xTerrainIntersection, int *yTerrainIntersection parameter, and just use the current master signature